sources/api/corndog/src/main.rs

```diff
@@ -107,6 +117,63 @@ where
     }
 }
 
+/// Generate the hugepages setting for defaults.
+fn generate_hugepages_setting() -> Result<()> {
+    let mut hugepages = "0".to_string();
+    // Check if customer has directly written to the nr_hugepage file.
+    if let Ok(existing_hugepages_value) =
+        fs::read_to_string(NR_HUGEPAGES_PATH_SYSCTL).context(error::ReadHugepagesFileSnafu)
+    {
+        hugepages = check_for_existing_hugepages(existing_hugepages_value);
+    }
+    if let Ok(is_efa_attached) = pciclient::is_efa_attached() {
+        if is_efa_attached {
+            // We will use [`num_cpus`] to get the number of cores for the compute.
+            hugepages = compute_hugepages_for_efa(num_cpus::get());
+        }
+    }
```

Comment:
This ends up clobbering the existing hugepages value if EFA is attached.

Reply:
Thanks for the nice catch. The refactoring kind of impacted the logic here. Just updated.